tests/api_tests/nodes/test_views.py

```diff
@@ -879,6 +879,514 @@ def test_filtering_node_with_non_bibliographic_contributor(self):
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['id'], self.user_two._id)
+
+        self.private_project.reload()
+        assert_in(self.user_two, self.private_project.contributors)
+
+    def test_adds_contributor_without_bibliographic_private_project_admin(self):
+        data = {
+            'id': self.user_two._id,
+        }
+        res = self.app.post_json(self.private_url, data, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+
+        self.private_project.reload()
+        assert_not_in(self.user_two, self.private_project.contributors)
+
+    def test_adds_admin_contributor_private_project_admin(self):
+        data = {
+            'id': self.user_two._id,
+            'permission': 'admin',
+            'bibliographic': True
+        }
+        res = self.app.post_json(self.private_url, data, auth=self.user.auth)
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['id'], self.user_two._id)
+
+        self.private_project.reload()
+        assert_in(self.user_two, self.private_project.contributors)
+        assert_equal(self.private_project.get_permissions(self.user_two), ['read', 'write', 'admin'])
+
+    def test_adds_write_contributor_private_project_admin(self):
+        data = {
+            'id': self.user_two._id,
+            'permission': 'write',
+            'bibliographic': True
+        }
+        res = self.app.post_json(self.private_url, data, auth=self.user.auth)
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['id'], self.user_two._id)
+
+        self.private_project.reload()
+        assert_in(self.user_two, self.private_project.contributors)
+        assert_equal(self.private_project.get_permissions(self.user_two), ['read', 'write'])
+
+    def test_adds_read_contributor_private_project_admin(self):
+        data = {
+            'id': self.user_two._id,
+            'permission': 'read',
+            'bibliographic': True
+        }
+        res = self.app.post_json(self.private_url, data, auth=self.user.auth)
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['id'], self.user_two._id)
+
+        self.private_project.reload()
+        assert_in(self.user_two, self.private_project.contributors)
+        assert_equal(self.private_project.get_permissions(self.user_two), ['read'])
+
+    def test_adds_invalid_permission_contributor_private_project_admin(self):
+        data = {
+            'id': self.user_two._id,
+            'permission': 'invalid',
+            'bibliographic': True
+        }
+        res = self.app.post_json(self.private_url, data, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+
+        self.private_project.reload()
+        assert_not_in(self.user_two, self.private_project.contributors)
+
+    def test_adds_none_permission_contributor_private_project_admin(self):
```

Comment:
This tests that the user is added, but not the effect of the "none" permission. Might be worth adding an assert?